tracer/src/Datadog.Trace/Util/ThrowHelper.cs

```diff
@@ -0,0 +1,71 @@
+// <copyright file="ThrowHelper.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+
+namespace Datadog.Trace.Util
+{
+    internal class ThrowHelper
```

Comment:
We should think about adding analyzers for throwing exceptions, at least for common cases `ArgumentNullException`. (not now, just for the backlog)

Reply:
I agree.